### DISCUSSION TITLE
Contextful keyword 'It'

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1291] [General]
'it' can be a keyword to refer to many things, it can let people type less effectively, here's an example
```
module UIOptionsSpecification
    let UIOptionsSpecification =
        let mutable ``Colorize items in iconic view`` = false
        {|
            toggleColorization = fun () ->
                ``Colorize items in iconic view`` <- not it//``Colorize items in iconic view``
        |}
```

**Comment by Happypig375** [comment]
So a self reference keyword in mutation expressions?

**Reply by Xyncgas** [reply]
For now yes
I can get another use case for you just let me check my notes.. the 'it' keyword can be use in record construction for referring to the variable before 'with' (if you see with you can use it)
